src/Microsoft.TemplateEngine.Cli/HelpAndUsage/TemplateInformationCoordinator.cs

```diff
@@ -368,6 +376,44 @@ private static New3CommandStatus DisplayInvalidParameterError(TemplateGroupMatch
             return New3CommandStatus.InvalidParamValues;
         }
 
+        private static void DisplayHintForOtherLanguages(INewCommandInput commandInput, TemplateResolutionResult resolutionResult)
+        {
+            if (resolutionResult.UnambiguousTemplateGroup == null || resolutionResult.UnambiguousTemplateGroupMatchInfo == null)
+            {
+                return;
+            }
+
+            if (resolutionResult.UnambiguousTemplateGroup.Languages.Count <= 1)
+            {
+                return;
+            }
+
+            string? preferredLanguage =
+                resolutionResult.UnambiguousTemplateGroupMatchInfo
+                    .TemplatesWithMatchingParametersForPreferredLanguage.First().GetLanguage();
+
+            List<string> supportedLanguages = new List<string>();
+            foreach (string? language in resolutionResult.UnambiguousTemplateGroup.Languages)
+            {
+                if (string.IsNullOrWhiteSpace(language))
+                {
+                    continue;
+                }
+                if (preferredLanguage == null || !language.Equals(preferredLanguage, StringComparison.OrdinalIgnoreCase))
```

Comment:
Nit: Null check isn't necessary.
```suggestion
                if (!language.Equals(preferredLanguage, StringComparison.OrdinalIgnoreCase))
```